source/Calamari/Kubernetes/Scripts/KubectlPowershellContext.ps1

```diff
@@ -134,17 +137,28 @@ function ConnectAzAccount {
 
 function SetupContext {
 	if($K8S_AccountType -ne "AzureServicePrincipal" -and [string]::IsNullOrEmpty($K8S_ClusterUrl)){
-		Write-Error "Kubernetes cluster URL is missing"
-		Exit 1
+    Write-Error "Kubernetes cluster URL is missing"
+    Exit 1
 	}
 
 	if([string]::IsNullOrEmpty($K8S_AccountType) -and [string]::IsNullOrEmpty($K8S_Client_Cert) -and $EKS_Use_Instance_Role -ine "true"){
-		Write-Error "Kubernetes account type or certificate is missing"
-		Exit 1
+	  if([string]::IsNullOrEmpty($Octopus_K8s_Pod_Service_Account_Token_Path) -and [string]::IsNullOrEmpty($Octopus_K8s_Server_Cert_Path)){
+	    Write-Error "Kubernetes account type or certificate is missing"
+    	Exit 1
+	  }
+		
+		$Octopus_K8s_Pod_Service_Account_Token = Get-Content -Path $Octopus_K8s_Pod_Service_Account_Token_Path
+		$Octopus_K8s_Server_Cert = Get-Content -Path $Octopus_K8s_Server_Cert_Path

```

Comment:
If `$Octopus_K8s_Server_Cert_Path` is an empty string, I suspect `Get-Content` will display an error. A check around this call to verify `$Octopus_K8s_Server_Cert_Path` is not empty would be a good idea.